File: app/src/main/java/com/orangepenguin/boilerplate/screens/userdetails/UserDetailsActivity.java
package com.orangepenguin.boilerplate.screens.userdetails;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;

import com.orangepenguin.boilerplate.BaseActivity;
import com.orangepenguin.boilerplate.R;
import com.orangepenguin.boilerplate.model.User;
import com.orangepenguin.boilerplate.util.ImageFetcher;

import javax.inject.Inject;
import javax.inject.Named;

import butterknife.BindView;

public class UserDetailsActivity extends BaseActivity<UserDetailsPresenter>
        implements UserDetailsView {

    private static final String USER_INTENT_PARAM = "USER_INTENT_PARAM";
    @Inject @Named("fitCenterCropWhitePlaceholder")
    ImageFetcher<String, ImageView> imageFetcher;
    @Inject UserDetailsPresenter presenter;
    @BindView(R.id.avatar_imageview) ImageView avatarImageView;
    @BindView(R.id.username_textview) TextView usernameTextView;
    @BindView(R.id.name_textview) TextView nameTextView;
    @BindView(R.id.company_textview) TextView companyTextView;
    @BindView(R.id.repos_textview) TextView reposTextView;
    private User user;

    public static Intent makeIntent(Context context, User user) {
        Intent intent = new Intent(context, UserDetailsActivity.class);
        intent.putExtra(USER_INTENT_PARAM, user);
        return intent;
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        // super.onCreate() has to be called *after* retrieving component. Android tries to reinitialize any Fragments
        // during super.onCreate() and those would rely on viewComponent already having been created.
        getViewComponent().inject(this);

        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_user_details);

        user = getIntent().getParcelableExtra(USER_INTENT_PARAM);
        presenter.initialize(savedInstanceState);
    }

    @Override
    public void onStart() {
        super.onStart();
        presenter.takeView(this);
        presenter.setUser(user);
    }

    @Override
    protected void onStop() {
        super.onStop();
        presenter.dropView();
    }

    @Override
    public void setUsername(String username) {
        usernameTextView.setText(username);
    }

    @Override
    public void setCompany(String company) {
        companyTextView.setText(company);
    }

    @Override
    public void setName(String name) {
        nameTextView.setText(name);
    }

    @Override
    public void setAvatarUrl(String avatarUrl) {
        imageFetcher.fetchAndSetImage(avatarUrl, avatarImageView);
    }

    @Override
    protected UserDetailsPresenter getPresenter() {
        return presenter;
    }
}

File: app/src/main/java/com/orangepenguin/boilerplate/screens/userdetails/UserDetailsActivity.kt
package com.orangepenguin.boilerplate.screens.userdetails

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import com.orangepenguin.boilerplate.BaseActivity
import com.orangepenguin.boilerplate.R
import com.orangepenguin.boilerplate.model.User
import com.orangepenguin.boilerplate.util.ImageFetcher
import kotterknife.bindView
import javax.inject.Inject
import javax.inject.Named

class UserDetailsActivity : BaseActivity<UserDetailsPresenter>(), UserDetailsView {
    @Inject
    @field:Named("fitCenterCropWhitePlaceholder")
    internal lateinit var imageFetcher: ImageFetcher<String, ImageView>
    @Inject override lateinit var presenter: UserDetailsPresenter

    //TODO: look at Kotlin synthetic properties
    private val ivAvatar: ImageView by bindView(R.id.avatar_imageview)
    private val tvUsername: TextView by bindView(R.id.username_textview)
    private val tvName: TextView  by bindView(R.id.name_textview)
    private val tvCompany: TextView by bindView(R.id.company_textview)
    private val tvRepos: TextView by bindView(R.id.repos_textview)
    private var user: User? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        // super.onCreate() has to be called *after* retrieving component. Android tries to reinitialize any Fragments
        // during super.onCreate() and those would rely on viewComponent already having been created.
        viewComponent.inject(this)

        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_user_details)

        user = intent.getParcelableExtra<User>(USER_INTENT_PARAM)
        presenter.initialize(savedInstanceState)
    }

    public override fun onStart() {
        super.onStart()
        presenter.takeView(this)
        presenter.user = user
    }

    override fun setUsername(username: String) {
        tvUsername.text = username
    }

    override fun setCompany(company: String) {
        tvCompany.text = company
    }

    override fun setName(name: String) {
        tvName.text = name
    }

    override fun setAvatarUrl(avatarUrl: String) {
        imageFetcher.fetchAndSetImage(avatarUrl, ivAvatar)
    }

    companion object {
        private val USER_INTENT_PARAM = "USER_INTENT_PARAM"

        fun makeIntent(context: Context, user: User): Intent {
            val intent = Intent(context, UserDetailsActivity::class.java)
            intent.putExtra(USER_INTENT_PARAM, user)
            return intent
        }
    }
}
